test/dotnet-new.Tests/GivenThatIWantANewCSApp.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using FluentAssertions;
+
+namespace Microsoft.DotNet.Tests
+{
+    public class GivenThatIWantANewCSApp : TestBase
+    {
+        [Fact]
+        public void When_NewtonsoftJson_dependency_added_Then_project_restores_and_runs()
+        {
+            var rootPath = Temp.CreateDirectory().Path;
+            var projectJsonFile = Path.Combine(rootPath, "project.json");
+
+            new TestCommand("dotnet") { WorkingDirectory = rootPath }
+                .Execute("new");
+            
+            AddProjectJsonDependency(projectJsonFile, "Newtonsoft.Json", "7.0.1");
```

Comment:
I think you should add code to the app that uses Newtonsoft. Last I tried you need an extra dependency to System.Runtime.Serialization.Primitives in order to actually use Newtonsoft.